config/provider.go

```diff
@@ -37,54 +37,54 @@ type Provider interface {
 
 	// A RegisterChangeCallback provides callback registration for config providers.
 	// These callbacks are nop if a dynamic provider is not configured for the service.
-	RegisterChangeCallback(key string, callback ConfigurationChangeCallback) error
+	RegisterChangeCallback(key string, callback ChangeCallback) error
 	UnregisterChangeCallback(token string) error
 }
 
-// ScopedProvider defines recursive interface of providers based on the prefix
-type ScopedProvider struct {
+// scopedProvider defines recursive interface of providers based on the prefix
+type scopedProvider struct {
 	Provider
 
 	prefix string
 }
 
 // NewScopedProvider creates a child provider given a prefix
-func NewScopedProvider(prefix string, provider Provider) *ScopedProvider {
-	return &ScopedProvider{provider, prefix}
-}
-
-func addPrefix(prefix, key string) string {
+func NewScopedProvider(prefix string, provider Provider) Provider {
 	if prefix == "" {
-		return key
+		return provider
 	}
 
+	return &scopedProvider{provider, prefix}
```

Comment:
nit: can we use named fields here?


```go
return &scopedProvider{
  Provider: provider,
  prefix: prefix,
}
```

It may seem overly verbose in this case but relying on implicit field orderings is frowned upon and can lead to subtle breakage if people move things around or add another field. Granted, the test suite should catch this, but I'd prefer to follow best practices regardless.